### ISSUE TITLE
→ Accessibility: Components | TextLink

### ISSUE THREAD

**Issue by yceballost** [Telefonica/mistica-design#1490]
#557 

The current link with the onPress prop is a button:

`<button type="button">Text</button>`
`role="link"` should be used to indicate the purpose of the component.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/link_role

## Tasks

- [x] Update specs
- [ ] Design documentation
- [x] [Android ticket](https://jira.tid.es/browse/ANDROID-13947)
- [x] [iOS ticket](https://jira.tid.es/browse/IOS-9503)
- [x] [Web ticket](https://jira.tid.es/browse/WEB-1623)
